File: tests/src/org.joda/joda-convert/2.2.4/src/test/java/org_joda/joda_convert/RenameHandlerTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_joda.joda_convert;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertSame;

import org.joda.convert.RenameHandler;
import org.junit.jupiter.api.Test;

public class RenameHandlerTest {
    @Test
    public void loadsRenameConfigurationResourcesFromClasspath() throws Exception {
        RenameHandler handler = RenameHandler.create(true);

        assertSame(RenameHandlerTest.class, handler.lookupType("example.LegacyName"));
        assertEquals(Status.NEW_STATUS, handler.lookupEnum(Status.class, "OLD_STATUS"));
    }

    public enum Status {
        NEW_STATUS
    }
}

File: tests/src/org.playframework.anorm/anorm_3/2.8.1/src/test/scala/org_playframework_anorm/anorm_3/RenameHandlerTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_playframework_anorm.anorm_3

import org.joda.convert.RenameHandler
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.Test

class RenameHandlerTest {
  @Test
  def loadsRenameConfigurationResourcesFromClasspath(): Unit = {
    val handler: RenameHandler = RenameHandler.create(true)

    assertSame(classOf[RenameHandlerTest], handler.lookupType("example.LegacyAnormName"))
    assertEquals(Thread.State.RUNNABLE, handler.lookupEnum(classOf[Thread.State], "OLD_RUNNABLE"))
  }
}
